src/plugins/VersionPlugin.ts

```diff
@@ -0,0 +1,93 @@
+import { AbstractPlugin, IMetaData } from '../plugin/AbstractPlugin'
+import PluginAPI from '../plugin/PluginAPI'
+import Translation from '@util/Translation';
+import Client from '../Client'
+
+const MIN_VERSION = '4.0.0';
+const MAX_VERSION = '99.0.0';
+
+const NAMESPACE_VERSION = 'jabber:iq:version'
+
+export default class VersionPlugin extends AbstractPlugin {
+
+   public static getId(): string {
+      return 'version';
+   }
+
+   public static getName(): string {
+      return 'Software Version';
+   }
+
+   public static getMetaData(): IMetaData {
+      return {
+         description: Translation.t('setting-version'),
+         xeps: [{
+            id: 'XEP-0092',
+            name: 'Software Version',
+            version: '1.1',
+         }]
+      }
+   }
+
+   constructor(pluginAPI: PluginAPI) {
+      super(MIN_VERSION, MAX_VERSION, pluginAPI);
+
+      let connection = pluginAPI.getConnection();
+
+      pluginAPI.addFeature(NAMESPACE_VERSION);
+
+      connection.registerHandler(this.onReceiveQuery, NAMESPACE_VERSION, 'iq');
+
+   }
+
+   //if name, version, os are empty it will be used as query for retrieve informations
+   public sendVersion(idstr: string, jid: string, name?: string, version?: string, os?: string): Promise<{}>
+   {
+      let iq = $iq({
+         type: (!name&&!version&&!os?'get':'result'),
+         to: jid,
+         id: idstr,
+         xmlns: 'jabber:client'
+      }).c('query', {
+         'xmlns': NAMESPACE_VERSION
+      });
+
+      if (name)
+      {
+          iq.c('name').t(name).up();
+      }
+
+      if (version)
+      {
+          iq.c('version').t(version).up();
+      }
+
+      if (os)
+      {
+          iq.c('os').t(os).up();
+      }
+
+      return this.pluginAPI.sendIQ(iq);
+   }
+
+   private onReceiveQuery = (stanza) => {
+        let OSName='Unknown OS';
+        if (navigator.appVersion.indexOf("Win")!==-1) OSName='Windows';
+        if (navigator.appVersion.indexOf("Mac")!==-1) OSName='MacOS';
+        if (navigator.appVersion.indexOf("X11")!==-1) OSName='UNIX';
+        if (navigator.appVersion.indexOf("Linux")!==-1) OSName='Linux';
+
+        let userAgent = navigator.userAgent || navigator.vendor || (<any>window).opera;
+
+        if (/android/i.test(userAgent)) {
+           OSName = 'Android';
+        }
+
+        if (/iPad|iPhone|iPod/.test(userAgent) && !window.MSStream) {
+            OSName = 'iOS';
+        }
+
+        this.sendVersion($(stanza).attr('id'),$(stanza).attr('from'),'JSXC',Client.getVersion(),OSName);
```

Comment:
We should only respond to contacts.

Reply:
How can i get the roster in a plugin?

Reply:
i would go over Client.getAccountManager()... loop all accounts and so on?

Reply:
Just try to get the contact from the plugin api (`api.getContact()`).

Reply:
thanks.. i should read the api source in future..

Reply:
Thats too much :joy: No problem. It's a pleasure to help.

Reply:
changed...